notebook/static/notebook/js/outputarea.js

```diff
@@ -659,9 +659,9 @@ define([
     var append_text = function (data, md, element) {
         var type = 'text/plain';
         var toinsert = this.create_output_subarea(md, "output_text", type);
+        data = utils.fixCarriageReturn(data);
```

Comment:
I'm not sure.
It seems more logical to me to first fix the carriage returns and then convert to HTML.

And it's done like this at a different place already.
At least it should be consistent, right?

Either way, the handling of CRs w.r.t. the terminal state machine isn't quite correct.
But I guess in practice, this has little relevance.